parcels/fieldset.py

```diff
@@ -92,8 +92,12 @@ def add_field(self, field):
     def check_complete(self):
         assert(self.U), ('U field is not defined')
         assert(self.V), ('V field is not defined')
+        ugrid = self.U.grid
         for g in self.gridset.grids:
             g.check_zonal_periodic()
+            if g is not ugrid and g.time_origin != 0:
```

Comment:
why is this `g is not ugrid` needed? I don't understand the rationale

Reply:
We want to re-shift time from all grids to the one of ugrid.
So no shift for ugrid.
But what happens if ugrid is independent of time and not the other grids? For the moment, it crashes (instead of giving non-sense result)